### PR TITLE
Feature/dtfs2 4120 applications page

### PR DESCRIPTION
**Pull request by davehlewis**
https://ukef-dtfs.atlassian.net/browse/DTFS2-4120
- adds new endpoint for combined deals query
- adds new schema to graphql
- removes filtering from dashboard
- updates other areas of dashboard to use combined query
- dashboard now uses sortable table
- changes flow for login - dashboard now the root (redirect), and login on explicit url. Avoids issues being logged in and still seeing login page

### CODE REVIEW

File: portal/server/api.js [hunk 1]
```diff
@@ -120,7 +133,7 @@ const updateDealName = async (id, newName, token) => {
 const updateDealStatus = async (statusUpdate, token) => {
   const response = await axios({
     method: 'put',
-    url: `${urlRoot}/v1/deals/${statusUpdate._id}/status`, // eslint-disable-line no-underscore-dangle
+    url: `${urlRoot}/v1/deals/${statusUpdate._id}/status`,
```
Comment (pricea1) [inline portal/server/api.js:136]: Is eslint passing without the es-lint-disable-no-underscrore-dangle? has it been globally configured?

**Review by pricea1 — APPROVED** [review]
LGTM but a couple of questions around eslint and the moving login to /login and dashboard to /. 
I know the dev who set this up discussed using /login and the dashboard at /. Just want to check there wasn't a business reason for doing it the way it was. 
/login and / for dashboard is the better choice in my opinion.

File: e2e-tests/gef/cypress/support/commands/login.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 const login = (credentials) => {
   const { username, password } = credentials;
 
-  cy.visit('/');
+  cy.visit('/login');
```
[the remark this answers is not in the document]
Comment (davehlewis) [reply e2e-tests/gef/cypress/support/commands/login.js:5]: yes - the journey made no sense having login on route, this avoids the issues with having login accessibly when already logged in. Not business reason not to do it that I know of an several people have brought it up along the way.

File: e2e-tests/portal/cypress/integration/journeys/maker/dashboard/deals/dashboard-table-data.spec.js [hunk 3]
```diff
@@ -4,139 +4,103 @@ const relative = require('../../../../relativeURL');
...
+        expect(text.trim()).equal('BSS/EWCS');
       });
 
-      const regexDateTime = /[\d][\d]\/[\d][\d]\/[\d][\d][\d][\d] [\d][\d]:[\d][\d]/;
-      row.updated().invoke('text').then((text) => {
-      // TODO - check formatting once formatting known
-        expect(text.trim()).to.match(regexDateTime);
+      status(id).invoke('text').then((text) => {
+        expect(text.trim()).equal('Draft');
       });
```
[the remark this answers is not in the document]
Comment (davehlewis) [reply e2e-tests/portal/cypress/integration/journeys/maker/dashboard/deals/dashboard-table-data.spec.js:66]: No, maker or checker no longer appear on dashboard

File: portal/server/api.js (hunk 1 above)
Comment (davehlewis) [reply portal/server/api.js:136]: yeah it's passing without, not made any changes to configuration. Running trough all tests with no problem

File: e2e-tests/portal/cypress/integration/journeys/maker/dashboard/deals/dashboard-table-data.spec.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (pricea1) [reply e2e-tests/portal/cypress/integration/journeys/maker/dashboard/deals/dashboard-table-data.spec.js:66]: 👍

File: e2e-tests/gef/cypress/support/commands/login.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (pricea1) [reply e2e-tests/gef/cypress/support/commands/login.js:5]: 👍

File: portal/server/api.js (hunk 1 above)
Comment (pricea1) [reply portal/server/api.js:136]: I've just checked and I added it to global config a couple of months ago :man_facepalming:

**Review by eddieleeper — APPROVED** [review]
LGTM!

**Pull request merged**